optaplanner-benchmark/src/main/java/org/optaplanner/benchmark/impl/PlannerBenchmarkRunner.java

```diff
@@ -169,13 +172,41 @@ private void warmUp() {
         logger.info("================================================================================");
     }
 
-    private void backupTerminationConfigs(Map<SolverBenchmarkResult, TerminationConfig> originalTerminationConfigMap) {
+    private Map<SolverBenchmarkResult, ConfigBackup> backupBenchmarkConfig(Map<ProblemBenchmarkResult, List<ProblemStatistic>> originalProblemStatisticMap) { // backup & remove stats, backup termination config
+        Map<SolverBenchmarkResult, ConfigBackup> configBackupMap = new HashMap<SolverBenchmarkResult, ConfigBackup>(plannerBenchmarkResult.getSolverBenchmarkResultList().size());
         for (SolverBenchmarkResult solverBenchmarkResult : plannerBenchmarkResult.getSolverBenchmarkResultList()) {
             TerminationConfig originalTerminationConfig = solverBenchmarkResult.getSolverConfig().getTerminationConfig();
-            if (!originalTerminationConfigMap.containsKey(solverBenchmarkResult)) {
-                originalTerminationConfigMap.put(solverBenchmarkResult, originalTerminationConfig);
+            ConfigBackup configBackup = new ConfigBackup(originalTerminationConfig);
+            for (SingleBenchmarkResult singleBenchmarkResult : solverBenchmarkResult.getSingleBenchmarkResultList()) {
+                List<PureSingleStatistic> originalPureSingleStatisticList = singleBenchmarkResult.getPureSingleStatisticList();
+                List<PureSingleStatistic> singleBenchmarkStatisticPutResult = configBackup.getPureSingleStatisticMap().put(singleBenchmarkResult, originalPureSingleStatisticList);
+                if (singleBenchmarkStatisticPutResult != null) {
+                    throw new IllegalStateException("SingleBenchmarkStatisticMap of ConfigBackup ( " + configBackup
+                            + " ) already contained key ( " + singleBenchmarkResult + " ) with value ( "
+                            + singleBenchmarkStatisticPutResult + " ).");
+                }
+                singleBenchmarkResult.setPureSingleStatisticList(Collections.<PureSingleStatistic>emptyList());
+
+                ProblemBenchmarkResult problemBenchmarkResult = singleBenchmarkResult.getProblemBenchmarkResult();
+                List<ProblemStatistic> originalProblemStatisticList = problemBenchmarkResult.getProblemStatisticList();
+                if (!originalProblemStatisticMap.containsKey(problemBenchmarkResult)) { // TODO: After Java 8, do Map#putIfAbsent
+                    List<ProblemStatistic> problemStatisticPutResult = originalProblemStatisticMap.put(problemBenchmarkResult, originalProblemStatisticList);
+                    if (problemStatisticPutResult != null && !originalProblemStatisticList.isEmpty()) {
```

Comment:
Citings put's javadoc: "**returns**    the previous value associated with key, or null if there was no mapping for key. (A null return can also indicate that the map previously associated null with key, if the implementation supports null values.)"

As far as I can tell, put behaves exactly like get regarding their return value (put additionally overrides the value for the key in the map, but that doesn't affect the return value).

Of course, you're right that the `!isEmpty()` check doesn't really make sense here: it will always be false, because we put a value in the line above, regardless of the putted value being null/non-null.